pkg/controller/nodeipam/ipam/cloud_cidr_allocator.go

```diff
@@ -99,6 +110,17 @@ func NewCloudCIDRAllocator(client clientset.Interface, cloud cloudprovider.Inter
 		err := fmt.Errorf("cloudCIDRAllocator does not support %v provider", cloud.ProviderName())
 		return nil, err
 	}
+
+	ipv4Found := isIP4(allocatorParams.ServiceCIDR) || isIP4(allocatorParams.SecondaryServiceCIDR)
+	ipv6Found := isIP6(allocatorParams.ServiceCIDR) || isIP6(allocatorParams.SecondaryServiceCIDR)
+
```

Comment:
comment explaining that this has been already validated so, is safe to assume that is not dual stack or ipv6 then it has to be ipv4

Reply:
I added such comments, PTAL.

Reply:
This code has changed, now taking into account the order of service CIDRs. The comments are no longer needed in fact, the code is self-explanatory. I only pointed out some validation that is performed elsewhere.